operator-framework-core/src/main/java/io/javaoperatorsdk/operator/Operator.java

```diff
@@ -92,6 +119,32 @@ public void close() {
     register(controller, null);
   }
 
+  /**
+   * Add a registration requests for the specified controller with this operator, overriding its
+   * default configuration by the specified one (usually created via {@link
+   * io.javaoperatorsdk.operator.api.config.ControllerConfigurationOverrider#override(ControllerConfiguration)},
+   * passing it the controller's original configuration. The effective registration of the
+   * controller is delayed till the operator is started.
+   *
+   * @param controller the controller to register
+   * @param configuration the configuration with which we want to register the controller, if {@code
+   *     null}, the controller's original configuration is used
+   * @param <R> the {@code CustomResource} type associated with the controller
+   * @throws OperatorException if a problem occurred during the registration process
+   */
+  public <R extends CustomResource> void register(
+      ResourceController<R> controller, ControllerConfiguration<R> configuration)
+      throws OperatorException {
+    synchronized (lock) {
+      if (!started) {
+        this.controllers.add(new ControllerRef(controller, configuration));
+      } else {
+        this.controllers.add(new ControllerRef(controller, configuration));
```

Comment:
The `ControllerRef` object is added to the queue in case a scenario like start, stop, start would be supported.